server/src/omtLinkProvider.ts

```diff
@@ -0,0 +1,197 @@
+     * declared imports ('module: moduleName'). For declared imports the target will be undefined.
+     * The declared import links can be resolved using the data and the `resolveLink` function.
+     * @param document the document containing OMT
+     */
+    provideDocumentLinks(document: TextDocument): DocumentLink[] {
+        // regular path links, with or without shorthands
+        const shorthands = this.contextPaths(document);
+        return findOMTUrl(document, (uri) => replaceStart(uri, shorthands));
+    }
+
+    /**
+     * Resolve the target of a DocumentLink using its data.
+     * @returns A string if the data could be related to a watched link target, such as a module declaration.
+     * Returns undefined when unable to resolve the link with the specified data.
+     * @param data the data from a document link
+     */
+    resolveLink(data?: DeclaredImportLinkData | unknown) {
+        if (data && isDeclaredImportLinkData(data)) {
+            return this.workspaceLookup.getModulePath(data.declaredImport.module);
+        }
+    }
+
+    /**
+     * Find all tsconfig files that apply to a given document
+     * and compile a list of paths annotations defined in those configurations.
+     * @param document the document that needs context.
+     */
+    private contextPaths(document: TextDocument) {
+        return new Map<string, string>(glob.sync('**/tsconfig**.json')
+            // we need to filter for same parent because another config may have some of the same paths
+            .filter(uri => document.uri.startsWith(uri.substr(0, uri.lastIndexOf('/'))))
+            .reduce((paths: [string, string][], uri: string) => {
+                try {
+                    const file = readFileSync(uri, 'utf8');
+                    const json = JSON.parse(file);
+                    if (json.compilerOptions && json.compilerOptions.paths) {
+                        // now make a path from the config folder to the keys value and map that
+                        for (const key in json.compilerOptions.paths) {
+                            const relPath = json.compilerOptions.paths[key].toString();
+                            const newPath = resolve(dirname(uri), relPath);
+                            paths.push([key.substr(0, key.lastIndexOf('/*')), newPath]);
+                        }
+                    }
+                } catch (e) {
+                    // need to catch this error for when there is a JSON parsing error for example
+                    // we can still continue with other files
+                    console.error(e);
+                }
+                return paths;
+            }, []));
+    }
+}
+
+/**
+ * returns all the text of a line in the provided document
+ * @param document A text document with at least [line] amount of lines
+ * @param line the line to be returned
+ */
+function getLine(document: TextDocument, line: number): string {
```

Comment:
because even though they are used by the class, they are not dependent on it's context. It's just something I have been trying to get used to in this project. The only exception is the `findOmtUrl` function, I will move that one back into the class. 
It seems to mostly have an advantage for classes that are created often.